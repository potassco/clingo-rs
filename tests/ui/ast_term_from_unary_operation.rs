use clingo::ast::*;
use clingo::*;

fn main() {
    let loc = Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term = symbolic_term(&loc, &sym).unwrap();
    let minus = UnaryOperator::Minus;
    let op = unary_operation(&loc, minus, term).unwrap();
    let term2: Term = op.into();
    drop(op);
    let _end = term2;
}
