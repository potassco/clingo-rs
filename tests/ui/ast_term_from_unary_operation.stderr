error[E0382]: use of moved value: `op`
  --> tests/ui/ast_term_from_unary_operation.rs:12:10
   |
10 |     let op = unary_operation(&loc, minus, term).unwrap();
   |         -- move occurs because `op` has type `clingo::ast::UnaryOperation<'_>`, which does not implement the `Copy` trait
11 |     let term2: Term = op.into();
   |                          ------ `op` moved due to this method call
12 |     drop(op);
   |          ^^ value used here after move
   |
note: `into` takes ownership of the receiver `self`, which moves `op`
  --> $RUST/core/src/convert/mod.rs
   |
   |     fn into(self) -> T;
   |             ^^^^
help: you can `clone` the value and consume it, but this might not be your desired behavior
   |
11 |     let term2: Term = op.clone().into();
   |                          ++++++++
